packages/js/product-editor/src/components/add-new-shipping-class-modal/add-new-shipping-class-modal.tsx

```diff
@@ -48,11 +112,36 @@ function ShippingClassForm( { onAdd, onCancel }: ShippingClassFormProps ) {
 						),
 					}
 				) }
+				onBlur={ getSlugSuggestion }
 			/>
-			<TextControl
+
+			<InputControl
 				{ ...getInputProps( 'slug' ) }
 				label={ __( 'Slug', 'woocommerce' ) }
+				onChange={ ( value: string ) => {
+					setPrevNameValue( '' ); // clean the previous name value.
+					getInputProps( 'slug' ).onChange( value );
+				} }
+				disabled={ isRequestingSlug }
+				help={ __(
+					'Set a custom slug or generate it by clicking the button.',
+					'woocommerce'
+				) }
+				prefix={
```

Comment:
Just curious why not using the `suffix` prop here instead since the Button component is aligned to the end of the input.

Reply:
Also there is a `InputControlSuffixWrapper` too.

Reply:
Indeed, this is the final version of a few suggestions we explored, particularly the one where we considered placing the button on the right side. I believe this approach could enhance the user experience.

I like this approach because, beyond my thoughts, it prioritizes the usage of the auto-generation slug over the custom one.

I was also tempted to focus on the description field right after the user clicked on the `Generate` button since we could induce that the user _really_ wanted to get the generated slug. I tried, and IMO, the UX gets smoother.

Reply:
Beautiful! If we are going to have it at the beginning of the input then it's ok to me.